gooddata-sdk/gooddata_sdk/catalog/data_source/service.py

```diff
@@ -80,3 +87,75 @@ def register_upload_notification(self, data_source_id: str) -> None:
 
     def get_declarative_data_sources(self) -> CatalogDeclarativeDataSources:
         return CatalogDeclarativeDataSources.from_api(self._layout_api.get_data_sources_layout())
+
+    def put_declarative_data_sources(
+        self,
+        declarative_data_sources: CatalogDeclarativeDataSources,
+        credentials_path: Optional[Path] = None,
+        test_data_sources: bool = False,
+    ) -> None:
+        if test_data_sources:
+            self.test_data_sources_connection(declarative_data_sources, credentials_path)
+        credentials = self._credentials_from_file(credentials_path) if credentials_path is not None else dict()
+        self._layout_api.put_data_sources_layout(declarative_data_sources.to_api(credentials))
+
+    def store_declarative_data_sources(self, path: Path) -> None:
+        data_sources = self.get_declarative_data_sources()
+        for data_source in data_sources.data_sources:
+            file_path = path / f"{data_source.id}.yaml"
+            with open(file_path, "w+", encoding="utf-8") as f:
+                yaml.safe_dump(data_source.to_api().to_dict(), f, indent=2, sort_keys=True)
+
+    @staticmethod
+    def _credentials_from_file(credentials_path: Path) -> dict[str, Any]:
+        if not os.path.isfile(credentials_path):
+            raise ValueError(f"There is no given file in the given path {credentials_path}")
+        with open(credentials_path, "r", encoding="utf-8") as f:
```

Comment:
By accident, a user can specify a path to a file, which is not in YAML format.
Try reading the YAML file and return a nice exception if it is not YAML.

Reply:
Do not force the suffix, it may be .yml as well and it means nothing.
Instead, please, encapsulate reading of file with try/except and return user friendly error if it fails.